### PR TITLE
Fix bug and add tests

### LINKED ISSUE #96: Implement test cases
Implemented test cases:
- `DeleteOrderCommand`
- `Quantity`
- `ItemName`
- `Description`
- `ItemContainsKeywordPredicate`
- `DeleteItemCommand`

Done, but needs refactoring to the `order` package:
- `AddressTest`
- `EmailTest`
- `NameTest`
- `OrderContainsKeywordsPredicate`
- `PhoneTest`

Unimplemented test cases:
- `AddItemCommand`
- `FindItemCommand`
- `ListItemsCommand`
- `EditOrderCommand`
- `AddOrderCommand` (revisit the commented tests)
- `ListOrdersCommand`
- `EditCommandParser`
- `AddItemCommandParser`
- `DeleteItemCommandParser`
- `FindItemCommandParser`

### LINKED ISSUE #124: Fix a bug in Mark Orders
Currently marking of orders targets the index based on filteredOrderList instead of sortedOrderList.

### PR DESCRIPTION
**Pull request by sohenze**
### Bug
---
Changed `MarkOrderCommand` to target `sortedList` instead of `filteredList`

### Tests
---
- Added test case for `DeleteOrderCommandTest`
- Added `timeCreated` field to `TypicalOrders`
- Added `SortOrderCommandTest`
- Renamed variables relating to items in `CommandTestUtil`
- Changed `ItemBuilder::withTags` to take `String` varargs as input instead for convenience (same implementation as AB3)
- Added components and tests for `EditItemCommandTest`

### CODE REVIEW

File: src/test/java/tracko/testutil/TypicalOrders.java [hunk 1]
```diff
@@ -67,25 +68,39 @@ public class TypicalOrders {
...
-            .withAddress("little tokyo").withItemList(List.of(DEFAULT_PAIR, PAIR_6)).build();
+            .withAddress("little tokyo")
+            .withTimeCreated(LocalDateTime.of(2022, 9, 25, 15, 30, 21))
+            .withItemList(List.of(DEFAULT_PAIR, PAIR_6)).build();
     public static final Order ORDER_7 = new OrderBuilder().withName("George Best")
             .withPhone("9482442").withEmail("anna@example.com")
-            .withAddress("4th street").withItemList(List.of(DEFAULT_PAIR, PAIR_7)).build();
+            .withAddress("4th street")
+            .withTimeCreated(LocalDateTime.of(2022, 10, 11, 12, 4, 9))
+            .withItemList(List.of(DEFAULT_PAIR, PAIR_7)).build();
```
Comment (melissaharijanto) [inline src/test/java/tracko/testutil/TypicalOrders.java]: I also added `timeCreated` to the TypicalOrders in my PR, but I used the `DEFAULT_TIME_CREATED` that's in one of the classes. I think your implementation of the different times is better than using just one default time, but I'm not sure if it's going to affect the tests in EditOrderCommand. Can you help me check (by pulling the latest merge)? So sorry for the trouble :"

**Review by melissaharijanto — APPROVED** [review]
Thanks Enze! I'll approve the PR, looks good to me!

File: src/test/java/tracko/testutil/TypicalOrders.java (hunk 1 above)
Comment (sohenze) [reply src/test/java/tracko/testutil/TypicalOrders.java:103]: Oh I see, actually `EditOrderCommand` tests have not been implemented yet, did you mean `EditItemCommand`? 

But it does affect the `SortOrderCommand` tests as the sort doesn't work if all orders have the exact same `timeCreated`. Is it alright if I change the current implementation to use different `timeCreated` for the orders used in `getTypicalOrders()`?

Also, if I'm not wrong, `OrderBuilder` already uses `DEFAULT_TIME_CREATED` when creating orders without specifying a time so there's no need to specify `DEFAULT_TIME_CREATED` actually

Comment (sohenze) [reply src/test/java/tracko/testutil/TypicalOrders.java]: Oh I see, actually doesn't Java CI already check if the tests work? So if all the checks pass it should be fine if I'm not wrong

File: src/test/java/tracko/logic/commands/DeleteOrderCommandTest.java [hunk 2]
```diff
@@ -78,6 +78,22 @@ public void execute_invalidIndexFilteredList_throwsCommandException() {
...
+        showOrderAtIndex(model, INDEX_FIRST);
+
+        Order orderToDelete = model.getSortedOrderList().get(INDEX_FIRST.getZeroBased());
+        DeleteOrderCommand deleteOrderCommand = new DeleteOrderCommand(INDEX_FIRST);
+
+        String expectedMessage = String.format(DeleteOrderCommand.MESSAGE_DELETE_ORDER_SUCCESS, orderToDelete);
+
+        Model expectedModel = new ModelManager(model.getTrackO(), new UserPrefs());
+        expectedModel.deleteOrder(orderToDelete);
+        showNoOrder(expectedModel);
```
Comment (Jiacheng-y) [inline src/test/java/tracko/logic/commands/DeleteOrderCommandTest.java:89]: I'm just curious, what does `showNoOrder(Model model)` do? I don't really understand the method :( 

There's nothing wrong with the code hahahaha I'm just trying to grapple with what the function does.

**Review by Jiacheng-y — APPROVED** [review]
Looks good, thanks Enze!

File: src/test/java/tracko/testutil/TypicalOrders.java (hunk 1 above)
Comment (melissaharijanto) [reply src/test/java/tracko/testutil/TypicalOrders.java]: Ah yep, I'm just slightly unsure, bc previously, the tests wouldn't pass bc timeCreated was not implemented in the OrderBuilder yet. I think can just merge and see; for the most part I think it should work out fine, I'm just not too sure haha

Comment (sohenze) [reply src/test/java/tracko/testutil/TypicalOrders.java]: Really? If I'm not wrong shouldn't `OrderBuilder` create orders with `DEFAULT_TIME_CREATED` by default?

File: src/test/java/tracko/logic/commands/DeleteOrderCommandTest.java (hunk 2 above)
Comment (sohenze) [reply src/test/java/tracko/logic/commands/DeleteOrderCommandTest.java:89]: It's actually implemented at the bottom of the file. It sets the filter predicate to false so no orders are shown, AB3's `DeleteCommand` tests had this as well so it's a refactor

Comment (Jiacheng-y) [reply src/test/java/tracko/logic/commands/DeleteOrderCommandTest.java:89]: does that mean that `expectedModel` does not contain any orders in its OrderList?

File: src/test/java/tracko/testutil/TypicalOrders.java (hunk 1 above)
Comment (melissaharijanto) [reply src/test/java/tracko/testutil/TypicalOrders.java]: Hmm I personally can't remember what the issue was exactly (with the test cases failing) aside from it being a difference in the `Order` String comparison, but if that's the case then I think the issue was with my `EditOrderCommand` bc it didnt copy over the time to the replacement order.

I think this change shouldn't affect the `EditOrderCommand` test cases then; I'm so sorry for the confusion!

Comment (sohenze) [reply src/test/java/tracko/testutil/TypicalOrders.java]: No worries, thanks for bringing it up! I'll double check after the merge to see if it's alright!

File: src/test/java/tracko/logic/commands/DeleteOrderCommandTest.java (hunk 2 above)
Comment (sohenze) [reply src/test/java/tracko/logic/commands/DeleteOrderCommandTest.java:89]: Nope, it just shows no orders in the UI. It's like using `findo` without a matching string so no orders show up. 

https://docs.oracle.com/javase/8/javafx/api/javafx/collections/transformation/FilteredList.html#setPredicate-java.util.function.Predicate-

File: src/test/java/tracko/testutil/TypicalOrders.java (hunk 1 above)
Comment (sohenze) [reply src/test/java/tracko/testutil/TypicalOrders.java:103]: Everything looks alright after the merge!

Comment (sohenze) [reply src/test/java/tracko/testutil/TypicalOrders.java]: Ok I see the problem now, basically, `Order::equals` doesn't actually check for equality in `timeCreated` but `Order::toString` has a `timeCreated` component to show the user at what time the order was created. However, the test actually checks for equality in the `CommandResult` as well, which uses `Order::toString`, which is why the different `timeCreated` causes different `CommandResults` 

I'm thinking of a suitable workaround and will update you when I'm done

File: src/test/java/tracko/logic/commands/DeleteOrderCommandTest.java (hunk 2 above)
Comment (Jiacheng-y) [reply src/test/java/tracko/logic/commands/DeleteOrderCommandTest.java:89]: ah I see, thank you :)

**Pull request merged**